artifacts/source_type.py

```diff
@@ -178,9 +185,22 @@ def __init__(self, keys=None):
     if not keys:
       raise errors.FormatError(u'Missing keys value.')
 
+    if not isinstance(keys, list):
+      raise errors.FormatError(u'keys must be a list')
+
+    for key in keys:
+      self.ValidateKey(key)
+
     super(WindowsRegistryKeySourceType, self).__init__()
     self.keys = keys
 
+  @classmethod
+  def ValidateKey(cls, key):
+    for prefix in cls.VALID_PREFIXES:
+      if key.startswith(prefix):
+        return
+    raise errors.FormatError(u'Not a supported registry key prefix, got: {}'.format(key))
```

Comment:
please have the style be consistent with the rest of the codebase, for now it uses {0:s} (or equiv)
{} => {0:s} (or equiv)